modules/videoio/src/cap_libcamera.cpp

```diff
@@ -0,0 +1,1001 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+/*
+ * Portions of this code are derived from https://github.com/kbarni/LCCV, which is licensed under the BSD 2-Clause "Simplified" License.
+ * The original code is Copyright (C) 2021, Raspberry Pi (Trading) Ltd..
+ * For the original BSD-licensed code, the following applies:
+ *
+ * Copyright (C) 2021, Raspberry Pi (Trading) Ltd.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ * 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*!
+ * \file cap_libcamera.cpp
+ * \file cap_libcamera.hpp
+ * 
+ * \author Xuanrui Zhu <sulingdie@gmail.com>
+ * 
+ * \author Jianru Xu <vegetableplanes@gmail.com>
+ *
+ * \author Zhian Chen <czabewin@gmail.com>
+ *
+ * \brief Use Libcamera to read/write video
+ */
+#include "precomp.hpp"
+#include "cap_libcamera.hpp"
+#include <mutex>
+
+namespace cv
+{
+
+std::string const &LibcameraCapture::CameraId() const
+{
+    return camera_->id();
+}
+
+void LibcameraCapture::OpenCamera()
+{
+
+    if (options_->verbose)
```

Comment:
verbose flag is not needed any more. CV_LOG_ messages are controlled by enviroment variables  OpenCV. See logging section of https://docs.opencv.org/4.x/d6/dea/tutorial_env_reference.html